soft_bfddpd/session.c

```diff
@@ -690,3 +692,47 @@ bfd_session_final_event(struct bfd_session *bs)
 	bfd_session_debug(bs, "final event");
 	bfd_session_dump(bs);
 }
+
+void bfd_session_set_state(struct bfd_session *bs, enum bfd_state_value state)
+{
+	if ((state >= STATE_ADMINDOWN) && (state <= STATE_UP)) {
+		if ((state == STATE_UP) && (bs->bs_state != STATE_UP)) {
+			bs->bs_up_count++;
+		} else if (((state == STATE_DOWN) || (state == STATE_ADMINDOWN)) &&
+				   (bs->bs_state != STATE_DOWN) && (bs->bs_state != STATE_ADMINDOWN)) {
+			bs->bs_down_count++;
+		}
+
+		bfd_session_debug(bs, "State changed %s -> %s",
+						  bfd_session_get_state_string(bs->bs_state),
```

Comment:
Indentation looks off, would you mind running clang-format to fix it please?

Reply:
Done, but other sections of the file were changed, too. I hope that is okay.